lighthouse-core/gather/gatherers/dobetterweb/js-vulnerable-libraries.js

```diff
@@ -0,0 +1,96 @@
+
+
+  let libraries = {};
+  for (let i in lighthouseJSLibs) {
+    try {
+      const result = lighthouseJSLibs[i].test(window);
+      if (result === false) continue;
+      if (npmMapper[i].npmPkgName) {
+        libraries[npmMapper[i].npmPkgName] = {"version": result.version};
+      }
+    } catch(e) {
+      console.log('Library Detector test for ' + i + ' failed:', e);
+    }
+  }
+  return libraries;
+}
+
+
+class JSVulnerableLibraries extends Gatherer {
+  /**
+   * @param {!Object} options
+   * @return {!Promise<!Array<!Object>>}
+   */
+  afterPass(options) {
+    const expression = `(function () {
+      return (${detectLibraries.toString()}());
+    })()`;
+    return options.driver
+      .evaluateAsync(expression)
+      .then(returnedValue => {
+        return new Promise((resolve, reject) => {
+          // need to mimic package.json for now
+          const jsonBody = '{"name": "lighthouse", "version": "1.0",' +
+            '"dependencies":' + JSON.stringify(returnedValue) + '}';
+          let vulnerabilities = [];
+          request.post({
+            headers: {
+              'content-type': 'application/json',
+              'Authorization': 'token 815130d4-940b-4252-b301-5ce28d734bf7'
```

Comment:
is there a rate limit on this token/who manages it? i don't know much about snyk 😄